packages/oauth-adapters/src/oauthAuthenticationAdapter.ts

```diff
@@ -52,27 +60,37 @@ function setOAuthTokenInRequest(
   );
 }
 
-function validateAuthorization(oAuthToken?: OAuthToken) {
+function validateAuthorization(oAuthToken?: OAuthToken, clockSkew?: number) {
   if (!isValid(oAuthToken)) {
     throw new Error(
       'Client is not authorized. An OAuth token is needed to make API calls.'
     );
   }
 
-  if (isExpired(oAuthToken)) {
+  if (isExpired(oAuthToken, clockSkew)) {
     throw new Error(
       'OAuth token is expired. A valid token is needed to make API calls.'
     );
   }
 }
 
-function isValid(oAuthToken: OAuthToken | undefined): oAuthToken is OAuthToken {
+export function isValid(
+  oAuthToken: OAuthToken | undefined
+): oAuthToken is OAuthToken {
   return typeof oAuthToken !== 'undefined';
 }
 
-function isExpired(oAuthToken: OAuthToken) {
-  return (
-    typeof oAuthToken.expiry !== 'undefined' &&
-    oAuthToken.expiry < Date.now() / 1000
-  );
+export function isExpired(oAuthToken: OAuthToken, clockSkew?: number) {
+  if (typeof oAuthToken.expiry === 'undefined') {
+    return false; // Expiry is undefined, token cannot be expired
```

Comment:
This line seems uncovered in the written unit tests @hamzamahmood

Reply:
The following test case is covering this scenario
``` ts
  it('should return false if expiry is undefined', () => {
    const token: OAuthToken = {
      accessToken: '1f12495f1a1ad9066b51fb3b4e456aee',
      tokenType: 'Bearer',
    };
    expect(isExpired(token)).toBe(false);
  });
```